sql/src/main/java/io/crate/execution/jobs/DistResultRXTask.java

```diff
@@ -99,6 +94,12 @@ public DistResultRXTask(Logger logger,
             () -> releaseListenersAndCloseContext(null)
         );
         this.consumer = rowConsumer;
+        if (bucketReceiverType.equals(BucketReceiverFactory.Type.MERGE_BUCKETS)) {
+            pageBucketReceiver = BucketReceiverFactory.createMergeBucketsReceiver(logger, nodeName, id, streamers, buckets,
```

Comment:
I don't think its a good pattern to create the bucket receiving component here, why not create it outside and pass it in? this would also avoid the factory and would decouple things more...

Reply:
Yes, that'd be ideal. It's inside the DistResultRXTask because it needs to receive the `kill` method reference

Reply:
So if it's outside, the DisResultRXTask needs the PageBucketReceiver in order to be built, and the PageBucketReceiver needs the instance `kill` method from the DistRestulRXTask (it's a circular dependency)

Reply:
hm yeah right, pitty.